src/hal/sensor/sensor.h

```diff
@@ -0,0 +1,17 @@
+
+#ifndef __SENSOR_HAL_H__
```

Comment:
Im not sure if I dont want to keep two separate header files and we can just have a single .cpp to implement them; since its not clear if wed want to combine them in all situations

Reply:
I'm okay either way. Just let me know.

Reply:
After discussion, leaving it as is for now.